src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamReadAsyncMemoryOverloadsTest : PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region No diagnostic - Analyzer
```

Comment:
Should I add "No diagnostic" unit tests for unsupported .NET versions? - Where the MemoryExtensions did not exist yet.

Reply:
Yes, that would be a good unit test, as long as you don't have to go through too many hoops to create a compilation where these types are not defined.

Reply:
Added. I didn't have to go through many hoops.